File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        n = 0
        res = []
        for i in nums:
            n = (n * 2 + i) % 5
            if n == 0:
                res.append(True)
            else:
                res.append(False)
        return res

File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        int n = 0;
        List<Boolean> res = new ArrayList();
        for (int x: nums) {
            n = (n * 2 + x) % 5;
            res.add(n==0);
        }
        return res;
    }
}